### DISCUSSION TITLE
Any reason why tooltip lags the cursor by a few frames?

### DISCUSSION THREAD

**Discussion by yay** [hecrj/iced#1334] [Q&A]
First of all, I have to say this project is awesome! Really happy to see things like iced and egui happening.

I'm wondering if there's a technical reason the tooltip looks like it's lagging the cursor by a few frames? (the footage below is slowed down to highlight the lag). I would expect a one frame lag.

https://user-images.githubusercontent.com/1799756/167048440-169bd071-46ab-49e0-b915-2d81bcaea10a.mov

**Comment by IvoWingelaar** [comment]
I want to add to this discussion that the tooltip is not the only widget that lags, it is a problem shared by *every* widget in the current version of `iced`. There appears to be some blocking *somewhere* going from user interaction to a visual update. I'm not sure if this happens in the main event loop of `iced_winit`, or maybe a vsync or a sync barrier between the shell in `iced_wgpu`, but it makes the GUI feel *very* sluggish when compared with a GUI implemented directly in GTK or another framework.

Another good example would be dragging panes in the `pane_grid` example, or resizing the separator between the panes:

https://user-images.githubusercontent.com/11836525/194183405-be04db34-3681-4118-98c5-bdf1d96b1e18.mp4

Compare that to dragging the separator in Thunar:

https://user-images.githubusercontent.com/11836525/194183483-ff2167d7-eb38-4275-9d81-ba5b50c86419.mp4

The difference is visually very large, but I'm completely stumped by what could be causing it.
